### PR TITLE
Fix Issue 18976 - Inconsistency in overload merging with aliases

### PR DESCRIPTION
**Pull request by JinShil**

### CODE REVIEW

**Comment by Geod24** [comment]
Restarting SemaphoreCI...

**Review by Geod24 — APPROVED** [review]

File: test/runnable/test18976.d [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// https://issues.dlang.org/show_bug.cgi?id=18976
```
Comment (WalterBright) [inline test/runnable/test18976.d]: This file should be in `compilable` not `runnable`. It would be nice to include the other case, too.

**Review by WalterBright — DISMISSED** [review]
The test suite will run faster if it's in `compilable`.

**Comment by RazvanN7** [comment]
The failure in semaphore seems unrelated. I observed it on different PRs too.

**Review by jacob-carlborg — APPROVED** [review]

**Comment by wilzbach** [comment]
```
src/object.d(562): Error: argument to __traits(getLinkage, core.memory.__unittest_L1028_C1.B) is not a declaration
src/core/memory.d(993): Error: template instance `object.destroy!(B)` error instantiating
src/core/memory.d(1044):        instantiated from here: __delete!(B)
src/core/memory.d(989): Error: template instance `object.destroy!(A)` error instantiating
src/core/memory.d(1078):        instantiated from here: __delete!(A)
src/core/memory.d(1155): Error: template instance `core.memory.__delete!(Object)` error instantiating
src/object.d(562): Error: argument to __traits(getLinkage, core.memory.__unittest_L1152_C1.C) is not a declaration
src/core/memory.d(993): Error: template instance `object.destroy!(C)` error instantiating
src/core/memory.d(1163):        instantiated from here: __delete!(C)
src/core/memory.d(989): Error: template instance `object.destroy!(I)` error instantiating
```

Could this error be a result from a recent commit to Druntime?
(we don't run the SemaphoreCI checks there)

**Comment by JinShil** [comment]
> Could this error be a result from a recent commit to Druntime?

Possibly, but what is different about Semphore CI.  Why would all other CIs pass for druntime, but not Semaphore CI.

**Comment by wilzbach** [comment]
SemaphoreCI tests all compilers (here gdc fails) + does a double bootstrapping from the current PR.

**Comment by JinShil** [comment]
@ibuclaw Can you offer any insight into the Semaphore CI failure?

**Comment by jacob-carlborg** [comment]
> Could this error be a result from a recent commit to Druntime?

There’s a fairly recent change to the `destroy` function, by Manu, which fixes/adds support for C++ classes [1]. That change relies on a recent change in DMD adds support for classes and interfaces to `__traits(getLinkage)`. 

[1] https://github.com/dlang/druntime/blob/master/src/object.d#L562

**Comment by ibuclaw** [comment]
DMD is also failing with the same error, the GDC test box just managed to fail first.

**Pull request merged**